processor/src/main/resources/org/mapstruct/ap/internal/model/IterableMappingMethod.ftl

```diff
@@ -121,7 +121,7 @@
         <#if resultType.fullyQualifiedName == "java.lang.Iterable">
             <@includeModel object=resultType.implementationType/>
         <#else>
-            <@includeModel object=resultType/>
+            <@includeModel object=resultType withoutBounds=true/>
```

Comment:
I'm not sure this is a good idea.. We are generating the methods ourselves because of nesting right? Should we even generate a method with a wildcard return type? See [here](http://www.angelikalanger.com/GenericsFAQ/FAQSections/ProgrammingIdioms.html#Should%20I%20use%20wildcards%20in%20the%20return%20type%20of%20a%20method?) for some details.. Just google "avoid return type wildcard". Of course.. we probably need to do something like this if the user decides to define a mapping method returning a wildcard.

Reply:
Don't have access to that. In any case we are delivering `Collection<? extends MyType>`. If you all at [`Collection#addAll(Collection)`](https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html#addAll-java.util.Collection-) it accepts `Collection<? extends E>` that is what the setter does actually too internally

Reply:
Fixed the link.. Perhaps I'm wrong.. I'll try to make an example